step-api/src/main/java/org/jenkinsci/plugins/workflow/steps/AbstractSynchronousStepExecution.java

```diff
@@ -6,7 +6,8 @@
 import static hudson.model.Result.ABORTED;
 
 /**
- * {@link StepExecution} that always executes synchronously.
+ * {@link StepExecution} that always executes synchronously. This API should be used for short-lived tasks that
+ * return almost instantly. For long-lived tasks use {@link AbstractSynchronousNonBlockingStepExecution}.
```

Comment:
Should clarify that long-lived tasks which do not need to run within a Java method call should use the more general `AbstractStepExecutionImpl`, and that `AbstractSynchronousNonBlockingStepExecution` is mainly for calling legacy Jenkins APIs which are potentially long-running and interruptible yet offer no asynchronous mode (for example because they block on a remoting call).